addons/mail/static/src/discuss/core/common/store_service_patch.js

```diff
@@ -15,6 +26,23 @@ const storeServicePatch = {
     sortMembers(m1, m2) {
         return m1.persona.name?.localeCompare(m2.persona.name) || m1.id - m2.id;
     },
+    /** @param {[number]} partnerIds */
+    async startChat(partnerIds) {
+        const partners_to = [...new Set([this.self.id, ...partnerIds])];
+        if (partners_to.length === 1) {
+            const chat = await this.joinChat(partners_to[0], true);
+            this.ChatWindow?.get({ thread: undefined })?.close();
+            chat.open();
+        } else if (partners_to.length === 2) {
+            const correspondentId = partners_to.find(
+                (partnerId) => partnerId !== this.store.self.id
+            );
+            const chat = await this.joinChat(correspondentId, true);
+            chat.open();
+        } else {
+            await this.createGroupChat({ partners_to });
```

Comment:
Why isn't the caller of `startChat` directly call `createGroupChat` instead?
It feels like the decision is made too low in the stack.

Reply:
We made `createGroupChat` before `startChat`. `startChat` is a good abstraction to plug with the code. `createGroupChat()` is best kept as an implementation detail.